server/src/main/java/org/elasticsearch/index/mapper/IgnoredFieldMapper.java

```diff
@@ -46,7 +54,7 @@
         static {
             FIELD_TYPE.setIndexOptions(IndexOptions.DOCS);
             FIELD_TYPE.setTokenized(false);
-            FIELD_TYPE.setStored(true);
+            FIELD_TYPE.setStored(false);
```

Comment:
This should be disabled only on indices created after this change (Version.V_8_0_0 for this PR). I guess you'll need two  `FIELD_TYPE`, one for legacy and one for the new configuration.